vms/platformvm/txs/executor/staker_tx_verification.go

```diff
@@ -524,24 +532,8 @@ func verifyAddPermissionlessValidatorTx(
 
 	var txFee uint64
 	if tx.Subnet != constants.PrimaryNetworkID {
-		primaryNetworkValidator, err := GetValidator(chainState, constants.PrimaryNetworkID, tx.Validator.NodeID)
-		if err != nil {
```

Comment:
The functionality change in this PR is adding the following code here:

```golang
if err == database.ErrNotFound {
	return fmt.Errorf(
		"%s %w of the primary network",
		validator.NodeID,
		ErrNotValidator,
	)
}
```

Reply:
to clarify, this should not change validity because the functional change is to provide a more specific error message and both errors are treated the same